### DISCUSSION TITLE
nuxt.refresh() does not remount components

### DISCUSSION THREAD

**Discussion by NickHatBoecker** [nuxt/nuxt#20953] [Questions (Nuxt 2)]
Hey there,

I want to fully reload the whole page, without losing my vuex store instance.
I found `nuxt.refresh()` which updates my `asynData`, but none of my components is remounted, so changes aren't shown.

Let's say I have a ModulePicker and a Textblock component. Based on the moduleType, a module is picked.

```js
{
    type: 'textblock',
    data: {
        text: 'This is my first text',
    },
}
```

I visit a page where this textblock module is rendered.
This is my index.vue:

```vue
<template>
    <div>
        <module-picker :data="modules" />
    </div>
</template>

<script>
export default {
    /**
     * {
     *     modules: [
     *         { type: String, text: String }
     *     ],
     * }
     */
    async asyncData (context) {
        const { data } = await httpGet('/some-api-url')

        return data
    }
}
</script>
```

Somewhere in the code I call `this.$nuxt.refresh()`.
So asyncData is reloaded and instead of the module response above, I have again a textblock module but with different text:

```js
{
    type: 'textblock',
    data: {
        text: 'This is another text after refresh',
    },
}
```

Via `console.log` statements I can understand that asyncData receives the correct data. But the changes never arrive in the Textblock component. Can I somehow trigger to recreate/remount all underlying components of `index.vue` oder ModulePicker component?

An alternative would be to route to a dummy page and reroute to the previous page, I guess?

Thanks in advance!

**Comment by maou-shonen** [comment]
[Official method for refreshing data in Nuxt.](https://nuxt.com/docs/api/utils/refresh-nuxt-data)

If you really want to destroy everything and rebuild...
```vue
<template>
  <div :key="{A reactive variable... will rebuild all DOM element when you change the value}">
    balabalabala...
    <NuxtPage />
    balabalabala...
  </div>
</template>
```
